bin/select_gfa

```diff
@@ -0,0 +1,146 @@
+    jj = np.fabs(delta_RA) < window_ra
+    jj = jj | ((delta_RA + 360.0) < window_ra)
+    jj = jj | ((360.0 - delta_RA) < window_ra)
+    jj = jj & (np.fabs(delta_dec)<window_dec)
+    return jj
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument('-o', '--output_dir', help='Path to write the outputs', type=str)
+parser.add_argument('-s', '--input_sweep', help='Path to the sweep files', type=str)
+parser.add_argument('-f', '--input_fiberassign', help='Path to fiberassign output files', type=str)
+parser.add_argument('-t', '--input_tiles', help='Input file with the tileid gfa files to be computed', type=str)
+
+args = parser.parse_args()
+
+# list tilefiles already available
+if args.input_fiberassign:
+    fiberassign_tilefiles = glob.glob(os.path.join(args.input_fiberassign,"tile*.fits"))
+    print(len(fiberassign_tilefiles), 'fiberassign tile files')
+else:
+    fiberassign_tilefiles = []
+
+if not os.path.isdir(args.output_dir):
+    os.makedirs(args.output_dir, exist_ok=True)
+
+# list sweep files to be used
+sweep_files = desitarget.io.list_sweepfiles(args.input_sweep)
+n_sweep = len(sweep_files)
+print(len(sweep_files), 'sweep files')
+
+# list of tiles to be computed
+desi_tiles = desimodel.io.load_tiles()
+if args.input_tiles is not None:
+    gfa_tiles = np.int_(np.loadtxt(args.input_tiles))
+    print(len(gfa_tiles), 'tiles to process')
+
+    # Check that all of them are known DESI tiles
+    unknown_tiles = ~np.in1d(gfa_tiles, desi_tiles['TILEID'])
+    if np.any(unknown_tiles):
+        nbad = np.count_nonzero(unknown_tiles)
+        print('{} unknown tiles: {}'.format(nbad, gfa_tiles[unknown_tiles]))
+
+    # Trim DESI tiles list to the requested subset
+    ii = np.in1d(desi_tiles['TILEID'], gfa_tiles)
+    desi_tiles = desi_tiles[ii]
+
+#load all sweep data
+sweep_data = []
+# n_sweep = 10
+for i in range(n_sweep):
+    sweep_file = sweep_files[i]
+    data = fitsio.read(sweep_file, columns=['RA', 'DEC', 'FLUX_R'])
+
+    #- Keep just mag>18
+    rfluxlim = 10**(0.4*(22.5-18))
+    ii = data['FLUX_R'] > rfluxlim
+    data = data[ii]
+
+    #- Faster for a small number of test tiles, but slower if using all tiles
+    # keep = np.zeros(len(data), dtype=bool)
+    # for tile in desi_tiles:
+    #     keep |= near_tile(data, tile['RA'], tile['DEC'])
+    # if np.any(keep):
+    #     sweep_data.append(data[keep])
+
+    sweep_data.append(data)
+
+    print('Loaded file {} out of {}'.format(i, n_sweep))
+
+all_sweep = np.concatenate(sweep_data, axis=0)
+
+print('There are {:.2f}M targets near tiles in the sweeps'.format(len(all_sweep)/1E6))
+
+#find IDs of targets on every individual tile
+
+for i in range(len(desi_tiles)):
+    tile_id = desi_tiles['TILEID'][i]
+    print('computing TILEID {:05d} on RA {:6.2f} DEC {:6.2f}'.format(tile_id, desi_tiles['RA'][i], desi_tiles['DEC'][i]))
+    
+    # select targets in a smaller window centered on tile
+    jj = near_tile(all_sweep, desi_tiles['RA'][i], desi_tiles['DEC'][i])
```

Comment:
I don't know if this comment belongs here or further up, but since we know the central coordinates and boundaries of all the sweeps (they're simple RA/Dec rectangles) can't we read the sweep file we need on-the-fly based on the tile we're considering?  I think this would scale much better than the current implementation.  Then the RA/Dec and magnitude cuts could be applied again using fitsio's `where` method, without reading in the whole file.

Another possibility (not for this PR, obviously) which we've talked about would be to repackage the sweeps into healpixels (for DESI purposes, not as part of the imaging survey DRs) and pre-compute the tiles that touch each sweep.  That would then make these kinds of target searches significantly faster.

Reply:
Those are good ideas, but I think we are not struggling for memory or time. We only use 1GB of RAM and processing a single takes ~0.5 seconds.